File: javas/version_1.0.0.js
import axios, { AxiosResponse } from 'axios';

console.log("🟢 Roblox Join Notifier is running!");

const UNIVERSE_ID: string = "7639730222";
const GAME_URL: string = "https://www.roblox.com/games/100339443227710/Sky";
const DISCORD_WEBHOOK: string = "webhook";

let lastMessageId: string | null = null;
let previousState: "online" | "offline" | null = null;
let lastPrintState: "online" | "offline" | null = null;

const now = (): string => {
    return new Date().toLocaleTimeString('en-US', { hour12: false });
};

const deleteLastMessage = async (): Promise<void> => {
    if (lastMessageId) {
        try {
            const deleteUrl: string = `${DISCORD_WEBHOOK}/messages/${lastMessageId}`;
            const response: AxiosResponse = await axios.delete(deleteUrl);
            if (response.status === 200 || response.status === 204) {
                console.log(`[${now()}] Deleted previous message.`);
            } else {
                console.log(`[${now()}] Failed to delete message: ${response.status} ${JSON.stringify(response.data)}`);
            }
        } catch (error: any) {
            console.log(`[${now()}] Error deleting message: ${error.message}`);
        }
        lastMessageId = null;
    }
};

const sendMessage = async (content: string): Promise<void> => {
    await deleteLastMessage();
    try {
        const response: AxiosResponse = await axios.post(`${DISCORD_WEBHOOK}?wait=true`, { content });
        if (response.status === 200 || response.status === 204) {
            console.log(`[${now()}] Webhook sent: ${content}`);
            try {
                lastMessageId = response.data.id || null;
                if (!lastMessageId) {
                    console.log(`[${now()}] Warning: Couldn't retrieve message ID.`);
                }
            } catch (error: any) {
                console.log(`[${now()}] Error parsing webhook response: ${error.message}`);
                lastMessageId = null;
            }
        } else {
            console.log(`[${now()}] Webhook error: ${response.status} ${JSON.stringify(response.data)}`);
        }
    } catch (error: any) {
        console.log(`[${now()}] Error sending webhook: ${error.message}`);
    }
};

const checkPlayers = async (): Promise<void> => {
    const url: string = `https://games.roblox.com/v1/games?universeIds=${UNIVERSE_ID}`;

    try {
        const response: AxiosResponse = await axios.get(url);
        if (response.status === 429) {
            console.log(`[${now()}] Rate limited. Waiting 30 seconds.`);
            await new Promise(resolve => setTimeout(resolve, 30000));
            return;
        }

        const count: number = response.data.data[0].playing;

        if (count > 0) {
            if (lastPrintState !== "online") {
                console.log(`[${now()}] Polled player count: ${count}`);
                lastPrintState = "online";
            }
        } else {
            if (lastPrintState !== "offline") {
                console.log(`[${now()}] Polled player count: 0`);
                lastPrintState = "offline";
            }
        }

        if (count > 0) {
            if (previousState !== "online") {
                await sendMessage(
                    `🎮 Someone just joined **Sky**!\nCurrently **${count}** player(s) in-game.\n🔗 <${GAME_URL}>`
                );
                previousState = "online";
            }
        } else {
            if (previousState !== "offline") {
                await sendMessage("No players are online.");
                previousState = "offline";
            }
        }
    } catch (error: any) {
        console.log(`[${now()}] Error: ${error.message}`);
    }
};

const main = async (): Promise<void> => {
    while (true) {
        await checkPlayers();
        await new Promise(resolve => setTimeout(resolve, 6000));
    }
};

main().catch(error => console.error(`[${now()}] Main loop error: ${error.message}`));

File: BuilderDump/version_1.0.0.py
import requests
import time

print("🟢 Roblox Join Notifier is running!")

UNIVERSE_ID = "7639730222"
GAME_URL = "https://www.roblox.com/games/100339443227710/Sky"
DISCORD_WEBHOOK = "webhook"

last_message_id = None
previous_state = None  # "online" or "offline""
last_print_state = None  # For console output xd

def now():
    return time.strftime('%H:%M:%S')

def delete_last_message():
    global last_message_id
    if last_message_id:
        try:
            delete_url = f"{DISCORD_WEBHOOK}/messages/{last_message_id}"
            response = requests.delete(delete_url)
            if response.status_code in [200, 204]:
                print(f"[{now()}] Deleted previous message.")
            else:
                print(f"[{now()}] Failed to delete message: {response.status_code} {response.text}")
        except Exception as e:
            print(f"[{now()}] Error deleting message: {e}")
        last_message_id = None

def send_message(content):
    global last_message_id
    delete_last_message()
    try:
        response = requests.post(DISCORD_WEBHOOK + "?wait=true", json={"content": content})
        if response.status_code in [200, 204]:
            print(f"[{now()}] Webhook sent: {content}")
            try:
                last_message_id = response.json().get("id")
                if last_message_id is None:
                    print(f"[{now()}] Warning: Couldn't retrieve message ID.")
            except Exception as e:
                print(f"[{now()}] Error parsing webhook response: {e}")
                last_message_id = None
        else:
            print(f"[{now()}] Webhook error: {response.status_code} {response.text}")
    except Exception as e:
        print(f"[{now()}] Error sending webhook: {e}")

def check_players():
    global previous_state, last_print_state

    url = f"https://games.roblox.com/v1/games?universeIds={UNIVERSE_ID}"

    try:
        response = requests.get(url)
        if response.status_code == 429:
            print(f"[{now()}] Rate limited. Waiting 30 seconds.")
            time.sleep(30)
            return

        response.raise_for_status()
        data = response.json()
        count = data['data'][0]['playing']

        if count > 0:
            if last_print_state != "online":
                print(f"[{now()}] Polled player count: {count}")
                last_print_state = "online"
        else:
            if last_print_state != "offline":
                print(f"[{now()}] Polled player count: 0")
                last_print_state = "offline"

        if count > 0:
            if previous_state != "online":
                send_message(f"🎮 Someone just joined **Sky**!\nCurrently **{count}** player(s) in-game.\n🔗 <{GAME_URL}>")
                previous_state = "online"
        else:
            if previous_state != "offline":
                send_message("No players are online.")
                previous_state = "offline"

    except Exception as e:
        print(f"[{now()}] Error: {e}")

while True:
    check_players()
    time.sleep(6)
